source/Calamari/Kubernetes/ResourceStatus/ResourceStatusReportWrapper.cs

```diff
@@ -0,0 +1,128 @@
+                                 !string.IsNullOrEmpty(variables.Get(SpecialVariables.GkeClusterName));
+            return hasClusterUrl || hasClusterName;
+        }
+
+        public CommandResult ExecuteScript(Script script, ScriptSyntax scriptSyntax,
+            ICommandLineRunner commandLineRunner,
+            Dictionary<string, string> environmentVars)
+        {
+            var result = NextWrapper.ExecuteScript(script, scriptSyntax, commandLineRunner, environmentVars);
+
+            if (!TryReadManifestFile(out var content))
+            {
+                return result;
+            }
+            
+            var customKubectlExecutable = variables.Get(SpecialVariables.CustomKubectlExecutable);
+            var deploymentTimeoutSeconds = variables.GetInt32(SpecialVariables.DeploymentTimeout) ?? 100;
+            var stabilizationTimeoutSeconds = variables.GetInt32(SpecialVariables.StabilizationTimeout) ?? 10;
+            var workingDirectory = Path.GetDirectoryName(script.File);
+            var definedResources = KubernetesYaml.GetDefinedResources(content).ToList();
+
+            if (definedResources.Count == 0)
+            {
+                return result;
+            }
+
+            foreach (var proxyVariable in ProxyEnvironmentVariablesGenerator.GenerateProxyEnvironmentVariables())
+            {
+                environmentVars[proxyVariable.Key] = proxyVariable.Value;
+            }
+            
+            var kubectl = new Kubectl(customKubectlExecutable, log, commandLineRunner, workingDirectory, environmentVars);
+            if (!kubectl.TrySetKubectl())
+            {
+                return new CommandResult(string.Empty, 1);
+            }
+
+            var completedSuccessfully = statusChecker.CheckStatusUntilCompletionOrTimeout(
+                definedResources, 
+                new CountdownTimer(TimeSpan.FromSeconds(deploymentTimeoutSeconds)), 
+                new CountdownTimer(TimeSpan.FromSeconds(stabilizationTimeoutSeconds)), 
+                kubectl);
+            
+            if (!completedSuccessfully)
+            {
+                return new CommandResult(string.Empty, 1, "Not all resources are deployed successfully within timeout");
+            }
+            
+            return result;
+        }
+
+        private bool TryReadManifestFile(out string content)
+        {
+            // TODO this won't handle configMaps defined together with a Deploy a Container step
+            var customResourceFileName =
+                variables.Get("Octopus.Action.KubernetesContainers.CustomResourceYamlFileName");
+            var knownFileNames = new[]
+            {
+                "secret.yml", customResourceFileName, "deployment.yml", "service.yml", "ingress.yml",
+            };
+            foreach (var file in knownFileNames)
+            {
+                if (!fileSystem.FileExists(file))
```

Comment:
Does this mean that if a user opts to load the config from a package that _doesnt_ match one of those names above, then it wont take effect?

Reply:
When customer selects a filename from a package, the name is set to the `Octopus.Action.KubernetesContainers.CustomResourceYamlFileName` variable. It is included in the list, so it should take effect.